mailpoet/lib/Cron/Workers/SendingQueue/SendingQueue.php

```diff
@@ -208,6 +214,17 @@ private function processSending(ScheduledTaskEntity $task, int $timer): void {
       return;
     }
 
+    // Pause task if sender domain requirements are not met
+    if (!$this->authorizedEmailsController->isSenderAddressValid($newsletter, 'sending')) {
+      $this->loggerFactory->getLogger(LoggerFactory::TOPIC_NEWSLETTERS)->info(
+        'pause task in sending queue due to sender domain requirements',
+        ['task_id' => $task->getId()]
+      );
+      $task->setStatus(ScheduledTaskEntity::STATUS_PAUSED);
+      $this->scheduledTasksRepository->flush();
+      return;
+    }
```

Comment:
This looks totally fine to me, but I'm curious to know what you think about the placement. We run this check before we start working on batches. Do we ever run into issues where the state changes in between batches and we should pause the sending but don't?

I think this would actually be much more likely to apply to the existing check above looking for deleted/trashed segments, and it's probably not relevant for this code since we're caching results for 24 hours with the transient.

On that note, what does that caching mean for users who have their campaigns paused and then validate their email address? Is there any chance they would have to wait, or can we always rely on [resumeSending](https://github.com/mailpoet/mailpoet/blob/994f46762d141387a06a205b933f5f96557cc887/mailpoet/lib/API/JSON/v1/Mailer.php#L86-L86) to break the cache through [checkAuthorizedEmailAddresses](https://github.com/mailpoet/mailpoet/blob/994f46762d141387a06a205b933f5f96557cc887/mailpoet/lib/Services/AuthorizedEmailsController.php#L125-L125)?

I'm mostly just curious if you think there's an argument for moving checks like these into `enforceSendingAndExecutionLimits` which runs on every batch.

Reply:
> Do we ever run into issues where the state changes in between batches and we should pause the sending but don't?

I don't think it would be common but it can happen. In that case the API would throw the error and we would pause the task. That's why I added the check for Paused tasks inside of the subscribers loop.
I can add a comment about it.

> On that note, what does that caching mean for users who have their campaigns paused and then validate their email address?

Validating the email address will reset the transient for Sender Domains because it calls `checkAuthorizedEmailAddresses`. 

> I'm mostly just curious if you think there's an argument for moving checks like these into enforceSendingAndExecutionLimits which runs on every batch.

I think it is a good idea and it would have made the work on this ticket much easier. 
I would do it on a separate ticket. We might not need to run all the checks every time `enforceSendingAndExecutionLimits` runs and we would need to be careful with the changes and the performance. This is also a critical part of the application, so taking the time for the refactor and adding tests would be the best.

Reply:
I added a comment when checking if the task has been paused when processing a batch in https://github.com/mailpoet/mailpoet/pull/5391/commits/c071b7f9b6c26a91b2905be6a42cfe58b7e39442

To move the checks to `enforceSendingAndExecutionLimits`, could you please create a maintenance ticket?